### PR TITLE
refactored pre-check

### PR DESCRIPTION
**Pull request by ammokhov**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by miparnisari** [comment]
I ran integration tests with this PR: 

image

I also ran them with this PR+removing the Read after the Create & Update operations:

image

File: aws-logs-metricfilter/src/main/java/software/amazon/logs/metricfilter/CreateHandler.java [hunk 1]
```diff
@@ -37,51 +29,30 @@
...
-                        return ProgressEvent.defaultFailureHandler(new CfnAlreadyExistsException(ResourceModel.TYPE_NAME, model.getPrimaryIdentifier().toString()), HandlerErrorCode.AlreadyExists);
-                    })
-            )
             .then(progress ->
                 proxy.initiate("AWS-Logs-MetricFilter::Create", proxyClient, model, callbackContext)
                     .translateToServiceRequest(Translator::translateToCreateRequest)
-                    .makeServiceCall(this::createResource)
+                    .makeServiceCall((r, c) -> createResource(model, r, c))
                     .progress())
             .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
```
Comment (wbingli) [inline aws-logs-metricfilter/src/main/java/software/amazon/logs/metricfilter/CreateHandler.java:42]: What is the ReadHandler doing here? is there any extra value by read?

Comment (ammokhov) [reply aws-logs-metricfilter/src/main/java/software/amazon/logs/metricfilter/CreateHandler.java]: removed

**Comment by ammokhov** [comment]
> I ran integration tests with this PR:
> 
> image
> 
> I also ran them with this PR+removing the Read after the Create & Update operations:
> 
> image

latest change results of 7 integ tests:
```
SUCCESSFUL in 2m 49s
```

**Comment by miparnisari** [comment]
```
============================================== short test summary info ==============================================
FAILED handler_update.py::contract_update_create_only_property - UnboundLocalError: local variable 'created_model'...
FAILED handler_update.py::contract_update_non_existent_resource - AssertionError: status should be FAILED
ERROR handler_delete.py::contract_delete_read - AssertionError: assert 'resourceModel' not in {'callbackDelaySecon...
ERROR handler_delete.py::contract_delete_list - AssertionError: assert 'resourceModel' not in {'callbackDelaySecon...
ERROR handler_delete.py::contract_delete_update - AssertionError: assert 'resourceModel' not in {'callbackDelaySec...
ERROR handler_delete.py::contract_delete_delete - AssertionError: assert 'resourceModel' not in {'callbackDelaySec...
ERROR handler_delete.py::contract_delete_create - AssertionError: assert 'resourceModel' not in {'callbackDelaySec...
===================== 2 failed, 7 passed, 1 skipped, 9 warnings, 5 errors in 152.70s (0:02:32) ======================
One or more contract tests failed
```

**Pull request merged**